pkg/controller/managedresources/controller.go

```diff
@@ -37,13 +39,17 @@ import (
 	"k8s.io/client-go/util/retry"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-// FinalizerName is the finalizer base name that is injected into ManagedResources.
-// The concrete finalizer is finally componed by this base name and the resource class.
-const FinalizerName = "resources.gardener.cloud/gardener-resource-manager"
+const (
+	// FinalizerName is the finalizer base name that is injected into ManagedResources.
+	// The concrete finalizer is finally componed by this base name and the resource class.
+	FinalizerName = "resources.gardener.cloud/gardener-resource-manager"
+	// ResourceManagerOperationAnnotation is an annotation that dictiates whether a resouces should be re-applied (reconciled)
+	// or simple ignore if it already exists.
+	ResourceManagerReconcileAnnotation = "resources.gardener.cloud/reconcile"
```

Comment:
Hmm, what about kube-addon-manager like approach like `resources.gardener.cloud/mode=EnsureExists` and `resources.gardener.cloud/mode=Reconcile`. In this way we will allow also the ignore feature and will allow users to deploy resources like `StorageClass`es and they will be able to customize them and gardener-resource-manager won't override the customizations?

/cc @rfranzke

Reply:
Ref https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/addon-manager#addon-manager

Reply:
With @zanetworker we agreed on `resources.gardener.cloud/ignore: <value>` annotation where a `value` is a truthy expression that pass `strconv.ParseBool(value)`.